partition_snapshot_reader.hh

```diff
@@ -287,9 +288,17 @@ class partition_snapshot_flat_reader : public flat_mutation_reader::impl, public
                         && !rt_less(pos, peek_range_tombstone().position())
                         && (_rt_stream.empty() || !rt_less(_rt_stream.peek_next().position(), peek_range_tombstone().position()))) {
                     range_tombstone rt = pop_range_tombstone();
-                    if (rt.trim(*_snapshot_schema,
-                                position_in_partition_view::for_range_start(ck_range),
-                                position_in_partition_view::for_range_end(ck_range))) {
+
+                    // To avoid copying ck_range even if we are not reversing, let's make this hack.
+                    std::optional<query::clustering_range> ck_range_opt;
+                    if (Reversing)
+                        ck_range_opt = query::reverse_clustering_range(ck_range);
+
+                    const query::clustering_range& ck_range_ref = ck_range_opt ? *ck_range_opt : ck_range;
+
+                    if (rt.trim(_schema,
```

Comment:
You're changing rt's schema from table domain to query domain without changing the emission site. This doesn't seem right, since the original domain didn't change.

Reply:
The emission site always assumed that the returned range tombstone is in the query domain (since before there was no distinction).

Reply:
This patch doesn't introduce the distinction, right? If not, then the old code was using the wrong schema in `rt.trim(*_snapshot_schema,` ?

Reply:
Perhaps it's an engineering issue on my side, maybe I should have not split the changes into multiple commits after all. Same issue as here https://github.com/scylladb/scylla/pull/9174#discussion_r712025502, and explanation: https://github.com/scylladb/scylla/pull/9174#discussion_r712101039